### ISSUE TITLE
Segfault when spinning up/down lots of containers

### ISSUE THREAD

**Issue by fasaxc** [docker/docker#22335]
**Output of `docker version`:**

```
Docker version 1.10.3, build 8acee1b
```

**Output of `docker info`:**

```
Containers: 262
 Running: 205
 Paused: 0
 Stopped: 57
Images: 5
Server Version: 1.10.3
Storage Driver: overlay
 Backing Filesystem: extfs
Execution Driver: native-0.2
Logging Driver: json-file
Plugins: 
 Volume: local
 Network: bridge null host
Kernel Version: 4.5.0-coreos-r1
Operating System: CoreOS 1010.1.0 (MoreOS)
OSType: linux
Architecture: x86_64
CPUs: 4
Total Memory: 3.469 GiB
Name: smc-kube-scale-01.c.unique-caldron-775.internal
ID: 6R53:BCSC:HN4W:ZCXI:DEP6:GPIK:2TB3:PM4I:7X4F:RRPE:D35A:X5EO
```

**Additional environment details (AWS, VirtualBox, physical, etc.):**

GCE, coreOS
4.5.0-coreos-r1 #2 SMP Thu Apr 7 03:52:11 UTC 2016 x86_64 Intel(R) Xeon(R) CPU @ 2.50GHz GenuineIntel GNU/Linux

**Steps to reproduce the issue:**

I was running scale tests with Kubernetes, spinning up 200 pods across 2 compute hosts, then killing the pods and starting again.

**Describe the results you received:**

Found this in the journal, if I'm reading it correctly, it looks like a segfault:
```
Apr 26 13:46:10 smc-kube-scale-01.c.unique-caldron-775.internal dockerd[1131]: unexpected fault address 0x1270
Apr 26 13:46:10 smc-kube-scale-01.c.unique-caldron-775.internal dockerd[1131]: fatal error: fault
Apr 26 13:46:10 smc-kube-scale-01.c.unique-caldron-775.internal dockerd[1131]: [signal 0xb code=0x1 addr=0x1270 pc=0x46fecd]
```
I'll attach/link to the full log.

**Describe the results you expected:**

No segfault.

**Additional information you deem important (e.g. issue happens only occasionally):**

Seen once or twice in maybe 10 runs.

**Comment by fasaxc** [comment]
[fault.txt](https://github.com/docker/docker/files/236784/fault.txt)

**Comment by cpuguy83** [comment]
Thanks, looks like this was fixed in https://github.com/opencontainers/runc/commit/c6cfce304b10513abd71252fddb57367f3c91673

Unfortunately this is not fixed in docker yet.

**Comment by cpuguy83** [comment]
Can you provde the full `docker version` details?

**Comment by mlaventure** [comment]
This is fixed in 1.11.1.
